test/HyperliquidInterchainToken.js

```diff
@@ -0,0 +1,297 @@
+
+        // Set up common values used across tests
+        provider = ethers.provider;
+        slot0 = await provider.getStorageAt(tokenAddress, 0);
+        deployerFromSlot0 = bytes32ToAddress(slot0);
+        deployerFromContract = await token.getDeployer();
+    });
+
+    describe('HyperliquidInterchainToken', () => {
+        it('Should calculate hardcoded constants correctly', async () => {
+            await expect(deployContract(owner, 'HyperliquidInterchainToken', [owner.address])).to.not.be.reverted;
+        });
+
+        it('should have the interchainTokenDeployer contract as deployer address after initialization', async () => {
+            expect(deployerFromContract).to.equal(ethers.constants.AddressZero);
+        });
+
+        it('should get the correct deployer address after updateDeployer', async () => {
+            await token.connect(owner).updateDeployer(deployer.address);
+            const newDeployer = await token.getDeployer();
+            expect(newDeployer).to.equal(deployer.address);
+        });
+
+        it('should allow only the service to update deployer', async () => {
+            await expect(token.connect(owner).updateDeployer(user.address)).to.not.be.reverted;
+            expect(await token.getDeployer()).to.equal(user.address);
+        });
+
+        it('should store deployer in slot 0', async () => {
+            expect(deployerFromSlot0.toLowerCase()).to.equal(deployerFromContract.toLowerCase());
+            // The deployer is initially AddressZero, so slot0 should be all zeros
+            expect(slot0).to.equal('0x0000000000000000000000000000000000000000000000000000000000000000');
+        });
+    });
+
+    describe('Bytecode checks [ @skip-on-coverage ]', () => {
+        it('Should preserve the same bytecode', async () => {
+            const contract = await ethers.getContractFactory('HyperliquidInterchainToken', owner);
+            const contractBytecode = contract.bytecode;
+
+            expect(contractBytecode).to.not.be.empty;
+        });
+    });
+
+    describe('Hyperliquid Deployer Tests', () => {
+        it('should get the deployer', async () => {
+            // Initially the deployer is AddressZero
+            expect(deployerFromContract).to.equal(ethers.constants.AddressZero);
+        });
+
+        it('should allow only the service to update the deployer', async () => {
+            const newDeployer = user.address;
+            await token.connect(owner).updateDeployer(newDeployer);
+            expect(await token.getDeployer()).to.equal(newDeployer);
+        });
+
+        it('should revert on updating deployer from non-service address', async () => {
+            const newDeployer = user.address;
+            await expect(token.connect(user).updateDeployer(newDeployer)).to.be.revertedWithCustomError(token, 'NotService');
+        });
+
+        it('should update the deployer for multiple contract instances', async () => {
```

Comment:
useless test, if it works on one instance, it'll work on a completely independent instance